### ISSUE TITLE
[BUG] Red and blue channels are swapped in vrdashboard

### ISSUE THREAD

**Issue by caseif** [ValveSoftware/SteamVR-for-Linux#570]
**Describe the bug**
The Steam overlay is not rendering colors correctly on the latest SteamVR beta (1.26.1). The issue does not exist on 1.25.7. As best I can tell, the red and blue color channels appear to be swapped indicating that some kind of image format mismatch is occurring.

**To Reproduce**
1. Opt into SteamVR beta
2. Load SteamVR home
3. Open SteamVR overlay

**Expected behavior**
The overlay should not swap color channels for rendering.

**System Information (please complete the following information):**
- Distribution: Arch
- SteamVR version: 1.26.1
- Steam client version: 2023/03/09 18:00:49
- Opted into Steam client beta?: Yes
- Graphics driver version:  Mesa 23.0.1
- [Gist for SteamVR System Information](https://gist.github.com/caseif/044d3220e9605e88b8d1607c3e9008e7)

**Screenshots**
![Screenshot_20230329_162123](https://user-images.githubusercontent.com/2825606/228661084-f91137e3-d3b4-4376-80ca-24d889efcf90.png)

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" section in their replies.

**Comment by c-x-berger** [comment]
Should the fact that some of the textures(?) used by vrdashboard here are getting displayed on the Knuckles controllers model (instead of battery amounts) be its own issue?

**Comment by caseif** [comment]
> Should the fact that some of the textures(?) used by vrdashboard here are getting displayed on the Knuckles controllers model (instead of battery amounts) be its own issue?

Yep, this is tracked as #571.

**Comment by ClearlyClaire** [comment]
Same here since last update (1.26.5).

**System Information:**

- Distribution: Debian testing
- SteamVR version: 1.26.5
- Steam client version: 1688685284
- Opted into Steam client beta?: Yes
- Graphics driver version: Mesa 22.3.6
- Gist for SteamVR System Information: https://gist.github.com/ClearlyClaire/4140a24c026a9655db87e01814210f44

**Comment by Kagukara** [comment]
How was this pushed to the non-beta. It was fine for me until the update on July 7 "[MAJOR UPDATE](https://store.steampowered.com/news/app/250820/view/6488045762500651005?l=english)".

**Comment by casKd-dev** [comment]
Same here: 

![image](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/30688640/17236087-3943-4635-87af-827ac7ecdc9e)

System info: https://gist.github.com/casKd-dev/c39266b15c2fd117c60e4a4f13ce424f
SteamVR info: https://gist.github.com/casKd-dev/ac2c275ae517ac8badc7453faa14ef0a

**Comment by sobersthebear** [comment]
Yup. I TOO have this issue, I will add that it seems to be restricted to the steamvr menus though. Everything else, like the boundary grid itself is normal.

KDE Neon 5.108 [fully updated]
Steamvr 1.26.6 [beta]
graphics nvidia 535.86.05
headset Index

**Comment by xriiitox** [comment]
Having this issue as well, again only in the dashboard.
Does not happen within games or SteamVR Home.

OS: EndeavourOS
SteamVR Version: 1.26.6
GPU: AMD Radeon RX 6600 (drivers up-to-date)
Headset: Quest 2 through ALVR

**Comment by purplebar0** [comment]
Whether or not it's related, I'm not entirely sure, but playing the Linux version of Half-Life: Alyx shows similar effects for save game previews. Since they're basically screenshots, I don't doubt that a component of SteamVR was used to generate it.

**Comment by Crunchylol** [comment]
Having this issue as well in the menu, game icons are also inverted. the store (old bigscreen) doesn't have this issue though, I suppose because it's not actually steamVR itself.

![Screenshot_20230810_115621](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/120267579/ad9eb030-99b8-4b02-914a-5ea15f0bc04e)

**System Information:**

Distribution: Fedora 38
SteamVR version: 1.26.7
Steam client version: 1690583737
Opted into Steam client beta?: No
Graphics card: NVIDIA GeForce GTX 1060 6GB
Gist for SteamVR system information: https://gist.github.com/Crunchylol/a777b4b89529f5b63c501a751cfdec75

**Comment by kisak-valve** [comment]
Hello, per "Fixed flipped red and blue channels on Linux textures." in the [SteamVR beta 1.27.2](https://store.steampowered.com/news/app/250820/view/6575871026909712810) update, please retest this issue.

**Comment by purplebar0** [comment]
Forgot to point it out, but the issue has been fixed in that update.

**Comment by casKd-dev** [comment]
kisak-valve ***@***.***> wrote:
> Hello, per "Fixed flipped red and blue channels on Linux textures." in the [SteamVR beta 1.27.2](https://store.steampowered.com/news/app/250820/view/6575871026909712810) update, please retest this issue.
Update has solved the issue for me. Thanks!

-- 
Alex D.
RedXen System & Infrastructure Administration
https://redxen.eu/

**Comment by caseif** [comment]
Appears to be fixed on my end as well, thanks!

**Comment by caseif** [comment]
This appears to have recurred in Steam VR 2.0.1.

@kisak-valve Should I reopen this issue, or create a new one?

**Comment by casKd-dev** [comment]
> Even worse, mines inverted AND upside down.

This behavior has indeed returned in the 2.0.2 beta. Can confirm.

**Comment by casKd-dev** [comment]
> > Even worse, mines inverted AND upside down.
> 
> This behavior has indeed returned in the 2.0.2 beta. Can confirm.

And it has been reversed together with the beta dashboard in 2.0.3 beta.

**Comment by vermeeren** [comment]
I have the problem too with Steam beta and SteamVR 2.0.10, Debian Mesa 23.2.1-1 and AMD Radeon Graphics (gfx1100, LLVM 16.0.6, DRM 3.54, 6.5.0-3-amd64.

**Comment by Joseph-DiGiovanni** [comment]
I'm also having this issue on 2.0.10. It started once my dashboard UI changed. Was looking fine with the old dashboard. This issue needs to be reopened for 2.0 it seems

**Comment by BestRazer** [comment]
Just installed SteamVR on my arch linux machine and I've got it too.

**Comment by caseif** [comment]
Doesn't seem to be resolved in 2.1.1.

**Comment by caseif** [comment]
Seems like the issue may have been fixed in the most recent Steam client beta. It's no longer occurring for me with SteamVR 2.0.10 when opted into the client beta. I'll wait for others to confirm before closing this issue again.

**Comment by MatrixFurry** [comment]
Seems to be fixed for me, too. I'm glad to see them working on this!

**Comment by BCNelson** [comment]
This seems to be back in 2.5.3

**Comment by TTimo** [comment]
This keeps breaking but should be fixed again in next beta update.
